src/constraints/battery_degradation.jl

```diff
@@ -114,77 +118,63 @@ function add_degradation(m, p; b="ElectricStorage")
-        When the battery degrades to 80% of purchased capacity it is replaced.
-        Multiple replacements could be necessary within the analysis period.
-        (The SOH is calculated for the analysis period, but not for multiple batteries.)
-        So we construct a cost as a function of months that accounts for the number of replacements.
-        (We use months instead of days to reduce the number of integer variables required).
-
-        The replacment cost in a given month is:
-        1. the maintenance_cost_per_kwh in (approximately) the 15th day of the month multiplied with
-        2. the number of replacements required given the first month that the battery must be replaced.
-        The number of months is analysis_years * 12.
-        The first month that the battery is replaced is determined by d_0p8, which is the integer 
-        number of days that the SOH is at least 80% of the purchased capacity.
-        We define a binary for each month and only allow one month to be chosen.
-        =#
         
-        # define d_0p8
         @warn "Adding binary and indicator constraints for 
-         ElectricStorage.degradation.maintenance_strategy = \"replacement\". 
-         Not all solvers support indicators and some are slow with integers."
-        @variable(m, soh_indicator[days], Bin)
-        @constraint(m, [d in days],
-            soh_indicator[d] => {SOH[d] >= 0.8*m[:dvStorageEnergy][b]}
-        )
-        @expression(m, d_0p8, sum(soh_indicator[d] for d in days))
-
-        # define binaries for the finding the month that battery must be replaced
-        months = 1:p.s.financial.analysis_years*12
-        @variable(m, bmth[months], Bin)
-        # can only pick one month (or no month if SOH is >= 80% in last day)
-        @constraint(m, sum(bmth[mth] for mth in months) == 1-soh_indicator[length(days)])
-        # the month picked is at most the month in which the SOH hits 80%
-        @constraint(m, sum(mth*bmth[mth] for mth in months) <= d_0p8 / 30.42)
-        # 30.42 is the average number of days in a month
-
-        #=
-        number of replacments as function of d_0p8
-         ^
-         |
-        4-    ------
-         |
-        3-          -------
-         |
-        2-                 -----
-         |
-        1-                      -------------------
-         |
-         ------|----|------|----|-----------------|->  d_0p8
-              N/5  N/4    N/3  N/2                N = 365*analysis_years
+        ElectricStorage.degradation.maintenance_strategy = \"replacement\". 
+        Not all solvers support indicators and some are slow with integers."
+        
+        @variable(m, soh_indicator[months], Bin) # track SOH levels, 1 if SOH >= 80%, 0 otherwise
+        @variable(m, bmth[months], Bin) # track which month SOH indicator drops to < 80%
+        @variable(m, 0 <= bmth_BkWh[months]) # track the kwh replacement value
+
+        M = 10*maximum(p.s.electric_load.loads_kw)  # the big M
```

Comment:
Note: Current big-M. This value seems to be giving good consistent results. Rationale is that a battery wont be larger than the peak load so 10x peak load is large enough. Even this value might be larger than expected. @zolanaj should we sweep across a range of big M values (i.e. 0.1x, 1.0x, 10x peak load) to fine tune this parameter?

Reply:
@rathod-b I checked the API and we never introduced a practical maximum size for batteries because they didn't have integer variables involved prior to the introduction of degradation.  

One heuristic would be power = 2x peak load, and energy = 12x peak load (i.e., allowing you to charge for half the day, and discharge for the other half of the day).  This precludes long-term storage systems, but we can probably include a user setting for long-term storage.

Related: I'll check our issues for importing big-M's from the API Julia code and add that issue if it's not there.

Reply:
big-M issue is here: https://github.com/NREL/REopt.jl/issues/114

Reply:
Actually, a system that has oversized PV to power the site load might need more.  just do 24x peak load for both power and energy, that should be enough.

Reply:
Updated big-M to be set based on max_kwh. Is it normal for changes in big-M to change the results? I see slight differences  in electric storage results.